File: VB/DxBlazorApplication1/Services/Implementations/InMemoryResourceService.vb
Imports DxBlazorApplication1.Models
Imports DxBlazorApplication1.Services.DataHelpers
Imports DxBlazorApplication1.Services.Interfaces
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace ServiceLayer.Implementations.InMemory

    Public Class InMemoryResourceService
        Implements IResourceService(Of Lecturer)

        Private ReadOnly resources As IList(Of Lecturer)

        Public Sub New()
            resources = ResourceDataHelper.GetResources()
        End Sub

        Public Function GetResourcesAsync() As Task(Of IList(Of Lecturer)) Implements IResourceService(Of Lecturer).GetResourcesAsync
            Return Task.FromResult(resources)
        End Function
    End Class
End Namespace

File: CS/DxBlazorApplication1/Services/Implementations/InMemoryResourceService.cs
using DxBlazorApplication1.Models;
using DxBlazorApplication1.Services.DataHelpers;
using DxBlazorApplication1.Services.Interfaces;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceLayer.Implementations.InMemory
{
    public class InMemoryResourceService : IResourceService<Lecturer>
    {
        private readonly IList<Lecturer> resources;

        public InMemoryResourceService()
        {
            resources = ResourceDataHelper.GetResources();
        }

        public Task<IList<Lecturer>> GetResourcesAsync()
        {
            return Task.FromResult(resources);
        }
    }
}
